pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -176,45 +176,68 @@ func (c *Cluster) Start() error {
 }
 
 func (c *Cluster) startMons() error {
-	// init the mons config
-	mons := c.initMonConfig(c.Count)
+	// init the mon config
+	existingCount, mons := c.initMonConfig(c.Count)
 
-	// Assign the pods to nodes
+	// Assign the mons to nodes
 	if err := c.assignMons(mons); err != nil {
 		return fmt.Errorf("failed to assign pods to mons. %+v", err)
 	}
 
-	// Start one monitor at a time
-	for i := 0; i < c.Count; i++ {
-		logger.Infof("ensuring mon %s (%s) is started", mons[i].ResourceName, mons[i].DaemonName)
-		endIndex := len(c.clusterInfo.Monitors)
-		if endIndex < c.Count {
-			endIndex++
+	if existingCount < len(mons) {
+		// Start the new mons one at a time
+		for i := existingCount; i < c.Count; i++ {
+			if err := c.ensureMonsRunning(mons, i, true); err != nil {
+				return err
+			}
 		}
-		logger.Infof("looping to start mons. i=%d, endIndex=%d, c.Size=%d", i, endIndex, c.Count)
-
-		// Init the mon IPs
-		if err := c.initMonIPs(mons[0:endIndex]); err != nil {
-			return fmt.Errorf("failed to init mon services. %+v", err)
+	} else {
+		// Ensure all the expected mon deployments exist, but don't require full quorum to continue
+		lastMonIndex := len(mons) - 1
+		if err := c.ensureMonsRunning(mons, lastMonIndex, false); err != nil {
+			return err
 		}
+	}
 
-		// save the mon config after we have "initiated the IPs"
-		if err := c.saveMonConfig(); err != nil {
-			return fmt.Errorf("failed to save mons. %+v", err)
-		}
+	logger.Debugf("mon endpoints used are: %s", FlattenMonEndpoints(c.clusterInfo.Monitors))
+	return nil
+}
 
-		// make sure we have the connection info generated so connections can happen
-		if err := writeConnectionConfig(c.context, c.clusterInfo); err != nil {
-			return err
-		}
+func (c *Cluster) ensureMonsRunning(mons []*monConfig, i int, requireAllInQuorum bool) error {
+	if requireAllInQuorum {
+		logger.Infof("creating mon %s", mons[i].DaemonName)
+	} else {
+		logger.Info("checking for basic quorum with existing mons")
+	}
 
-		// Start the deployment
-		if err := c.startDeployments(mons[0:endIndex], i); err != nil {
-			return fmt.Errorf("failed to start mon pods. %+v", err)
-		}
+	// Look at the number of mons already existing
+	expectedMonCount := len(c.clusterInfo.Monitors)
```

Comment:
The naming `expectedMonCount` is a bit misleading. To me `c.Count` is the expected monitor count. Isn't `len(c.clusterInfo.Monitors)` more like `currentMonitorCount`?

Reply:
`expectedMonCount` is the number of mons after this call. Since mons are added one at a time, it takes several calls to this method in a new orchestration before `c.Count` is reached. In the next line, hopefully the comment clarifies, but I'll clarify this comment here.